--- a/bin/build_addon
+++ b/bin/build_addon
@@ -33,16 +33,18 @@ else
   echo "Unknown target machine type.  This kit will only build ReadyNAS Padre, x86, and ARM platforms."
   exit 1
 fi
-if ! [ -d "../bin" -a -d "../template" -a -d "files" ]; then
+prog_dir=${0%/*}
+prog_dir=${prog_dir}${prog_dir:+/}
+if ! [ -d "${prog_dir:-.}" -a -d "${prog_dir:-.}/../template" -a -d "files" ]; then
   echo "This doesn't appear to be a valid Addon SDK base directory."
   exit 1
 fi
 
-file=$(ls *.xml | grep -v "_" | head -1)
-TAG=$(../bin/xmlutil.$harch $file /addon/name | awk -F'!!' '{ print $2 }')
-FRIENDLY_NAME=$(../bin/xmlutil.$harch $file /addon/friendly_name | awk -F'!!' '{ print $2 }')
+file=$(ls *.xml | egrep -v "(_AVAILABLE|_CURRENT).xml" | head -1)
+TAG=$(${prog_dir}xmlutil.$harch $file /addon/name | awk -F'!!' '{ print $2 }')
+FRIENDLY_NAME=$(${prog_dir}xmlutil.$harch $file /addon/friendly_name | awk -F'!!' '{ print $2 }')
 NAME=$(echo $FRIENDLY_NAME | sed 's/ //g')
-VERS=$(../bin/xmlutil.$harch $file /addon/version | awk -F'!!' '{ print $2 }')
+VERS=$(${prog_dir}xmlutil.$harch $file /addon/version | awk -F'!!' '{ print $2 }')
 START=/etc/frontview/addons/bin/$TAG/start.sh
 STOP=/etc/frontview/addons/bin/$TAG/stop.sh
 TIME=$(date +%s)
@@ -93,6 +95,8 @@ chmod +x files/etc/frontview/addons/ui/${TAG}/${TAG}_HANDLER.pl
 cp start.sh stop.sh running.sh files/etc/frontview/addons/bin/${TAG}/
 (cd files/etc/frontview/addons/bin/${TAG}/; chmod +x start.sh stop.sh running.sh)
 
+[ -f ${TAG}.jpg ] && cp ${TAG}.jpg files/etc/frontview/addons/ui/${TAG}/
+
 if [ -f ${TAG}_service ]; then
   cp ${TAG}_service files/etc/frontview/addons/bin/${TAG}/
   chmod +x files/etc/frontview/addons/bin/${TAG}/${TAG}_service
@@ -101,9 +105,11 @@ fi
 cp ${TAG}.conf files/etc/frontview/apache/addons/${TAG}.conf
 
 cd files || error 
-tar cfz ../files.tgz ./* || error
+FAKEROOT=$(command -v fakeroot)
+$FAKEROOT tar cfz ../files.tgz ./* || error
 cd ..
-tar cf ${NAME}_${VERS}.tar install.sh remove.sh addons.conf files.tgz || error
+chmod +x install.sh remove.sh
+$FAKEROOT tar cf ${NAME}_${VERS}.tar install.sh remove.sh addons.conf files.tgz || error
 
 SIZE=`stat -t ${NAME}_${VERS}.tar | awk '{ print $2 }'`
 MD5=`md5sum ${NAME}_${VERS}.tar | awk '{ print $1 }'`
